File: parking-domain/src/main/java/com/chatchatabc/parking/domain/model/ParkingRole.java
package com.chatchatabc.parking.domain.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@Entity
@Table(name = "parking_roles")
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ParkingRole extends BaseEntity {
    @Column
    private String name;
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/service/domain/model/ParkingRole.kt
package com.chatchatabc.service.domain.model

import jakarta.persistence.*
import lombok.Data

enum class ParkingRoleNames {
    ROLE_OWNER,
    ROLE_MANAGER,
    ROLE_GUARD
}

@Data
@Entity
@Table(name = "parking_roles")
open class ParkingRole(name: String) {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    open lateinit var id: String

    @Column(unique = true)
    open var name: String = name
}